### ISSUE TITLE
Recent update to txfunc.py seems broken

### ISSUE THREAD

**Issue by talumbau** [open-source-economics/OG-USA#240]
This recent commit:

https://github.com/open-source-economics/OG-USA/commit/df1abfeaa9784ef590907a3bed3fc7d377d62621

made significant changes to `txfunc.py`. In fact, the diff was so great that GitHub does not show the diff via the diffing tool online. I wonder if this was an intentional change. This new `txfunc.py` does not work for me. When I run the code with `age_specific=False`, the error I get is:

```
Traceback (most recent call last):
  File "./run_reforms.py", line 86, in <module>
    run_micro=run_micro)
  File "./run_reforms.py", line 46, in run_micro_macro
    runner(**kwargs)
  File "../execute_small.py", line 37, in runner
    start_year=user_params['start_year'], reform=reform, guid=guid)
  File "/Users/talumbau/miniconda3/envs/ospcdyn/lib/python2.7/site-packages/ogusa-0.5.5+77.g14bbadb.dirty-py2.7.egg/ogusa/txfunc.py", line 1565, in get_tax_func_estimate
  File "/Users/talumbau/miniconda3/envs/ospcdyn/lib/python2.7/site-packages/ogusa-0.5.5+77.g14bbadb.dirty-py2.7.egg/ogusa/txfunc.py", line 1191, in tax_func_estimate
UnboundLocalError: local variable 'df_minobs' referenced before assignment
```

This appears to be a correct error. When I look at the current code, I see that `df_minobs` is defined, but only if `age_specific` is `True`.

https://github.com/open-source-economics/OG-USA/blob/master/Python/ogusa/txfunc.py#L1181

Otherwise, `df_minobs` is not defined. However, in line [1191](https://github.com/open-source-economics/OG-USA/blob/master/Python/ogusa/txfunc.py#L1191), the value of `df_minobs` is needed. I am running the code with `age_specific=False` so I hit this error.

This underscores the need to stand up the test harness for longer running tests, since we don't run this code with every PR.

@rickecon can you confirm that what I've written above agrees with your inspection of the code and comment on the commit. It looks to me like this `txfunc.py` can only work with `age_specific=True`. Is that right?

**Comment by jdebacker** [comment]
I'm working on this now.  Will also allow for analytical_mtrs=True.

cc @rickecon

**Comment by rickecon** [comment]
I'll try to join in this morning as well. Those changes to `txfunc.py` were big and deliberate. I thought we had all the kinks worked out with `age_specific=False`. But why do we want to get `analytical_mtrs=True` working? Nobody wants that right now. Let's just get` age_specific=False` working.

**Comment by jdebacker** [comment]
Because it's low cost- just need to calc the derivatives by hand and enter in tax.py.

**Comment by rickecon** [comment]
OK. I will confirm the analytical derivatives with you. @jdebacker

**Comment by jdebacker** [comment]
Here's what I'm getting for an analytical derivative:
![mtrx](https://cloud.githubusercontent.com/assets/10715924/18391053/a11777e2-767a-11e6-8c13-3057e2c15e56.png)

Let me know if you get the same @rickecon

**Comment by rickecon** [comment]
I am close to your @jdebacker solution but not quite there. I got the same first equation as you.
[Eq1.pdf](https://github.com/open-source-economics/OG-USA/files/464231/Eq1.pdf)
Define the tax function in the following way:
[Eq2.pdf](https://github.com/open-source-economics/OG-USA/files/464253/Eq2.pdf)
Then,
[Eq3.pdf](https://github.com/open-source-economics/OG-USA/files/464280/Eq3.pdf)
So the analytical marginal tax rate on labor income defined above is the following:
[Eq4.pdf](https://github.com/open-source-economics/OG-USA/files/464281/Eq4.pdf)
Analogously, I get the following analytical marginal tax rate on capital income:
[Eq5.pdf](https://github.com/open-source-economics/OG-USA/files/464277/Eq5.pdf)

**Comment by jdebacker** [comment]
Thanks @rickecon, I was missing a shift parameter.  Updated equation for MTRx (MTRy analogous):
![mtrx_v2](https://cloud.githubusercontent.com/assets/10715924/18393503/9d1a8810-7683-11e6-82bb-44ebe410700e.png)

**Comment by jdebacker** [comment]
3rd try is a charm?  Forgot that it's total income multiplied by \tau(x,y):

![mtrx_v3](https://cloud.githubusercontent.com/assets/10715924/18394056/08e52e22-7686-11e6-8882-188cb77d0af9.png)

**Comment by rickecon** [comment]
@jdebacker . That's right. Just fix the first line so that it has (x+y) instead of just x.

**Comment by jdebacker** [comment]
Fixed via PR #241
